modules/oadp-mcg-performance-tuning.adoc

```diff
@@ -0,0 +1,139 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="oadp-mcg-performance-tuning_{context}"]
+= Performance tuning guide for Multicloud Object Gateway
+
+The default configuration of {rh-storage} for Multicloud Object Gateway is optimized for low resource consumption and not performance.
+
+
+* Large files: In the case of large files, the metadata-to-data ratio is low. Increasing the resources for the endpoints (memory and CPU) and the number of endpoints would be the first thing to do. In the case of namespace buckets, only memory would be sufficient. 
++
+The CPU is important mainly for data buckets, where the endpoints use the CPU for encryption and deduplication.
+
+* Small objects: In the case of small objects, the metadata-to-data ratio is high. For data buckets, this means a high level of resource involvement from the core and database. Increasing those pods resources would be the first step.
++
+Endpoints memory will probably not be under pressure if the core and database respond quickly. If they do not respond quickly enough, then pressure will build, and endpoints will eventually be under pressure. In this case, you should increase both core and database, with more emphasis on the database itself.
+
+* A high number of configuration entities, such as a large number of buckets or accounts: This would also point to the database and core, with more emphasis on the core than the database.
+
+When using namespace buckets, increasing the endpoint's memory and the database's memory and CPU would be the first step.
+
+*  MCG database resources: You need to increase CPU and memory based on the workload characteristics.
+
+* MCG auto-scales the minimum and maximum size: This would improve the response to peaks, but it also has a delay until it reacts. Therefore, it is important to set both the minimum and maximum size.
+
+* MCG Core resources: You need to increase CPU and memory based upon the workload characteristics.
+
+* Ensure you connect to the NooBaa endpoint by using its service address, link:https://s3.openshift-storage.svc[https://s3.openshift-storage.svc] or link:http://s3.openshift-storage.svc[http://s3.openshift-storage.svc] since this connects directly to the NooBaa endpoints.
+
+
+[id="mcg-auto-scaling_{context}"]
+== Adjust the auto-scaling
+
+* Adjust the auto-scaling with a command like this:
+
++
+[source,yaml]
+----
+oc patch -n openshift-storage storagecluster ocs-storagecluster \
+    --type merge \
+    --patch '{"spec": {"multiCloudGateway": {"endpoints": {"minCount": 3,"maxCount": 10}}}}'
+----
+
++
+This would set the NooBaa Endpoint Horizontal Pod Autoscaling to deploy at least 3 Pods and scale up to 10 Pods when needed. The default is to deploy at least 1 Pod and at most 2 Pods.
+
+
+[id="mcg-tuining-core-database-resources_{context}"]
+== Tuning MCG core and database resources
+
+* Tuning MCG core and database resources can be done via the storage cluster CR.
```

Comment:
```suggestion
+
This would set the NooBaa Endpoint Horizontal Pod Autoscaling, with the aim of automatically scaling the workload to match demand, to deploy at least 3 pods and scale up to 10 pods when needed. The default is to deploy at least 1 pod and at most 2 pods.


[id="mcg-tuining-core-database-resources_{context}"]
== Tuning MCG core and database resources

* Tuning MCG core and database resources by using the storage cluster Custom Resource (CR).
```